contracts/BridgeProxy.sol

```diff
@@ -0,0 +1,118 @@
+//SPDX-License-Identifier: UNLICENSED
+
+pragma solidity ^0.8.20;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "./lib/Pausable.sol";
+import "./SharedStructs.sol";
+import "./lib/BoolTokenTransfer.sol";
+import "./access/AdminRole.sol";
+import "./access/BridgeRole.sol";
+
+contract BridgeProxy is Initializable, Pausable, BridgeRole {
```

Comment:
> I would rename this. `BridgeProxy` implies that it's a proxy for the bridge, eg. before the Bridge. But it's a component behind it. Maybe SCExecutionProxy or something like that?

I totally agree with you but I went with BridgeProxy to keep it consistent with the BridgeProxy contract on the MultiversX side. My idea was to maintain symmetry between the contracts on both chains.

If you want, we can change the names for both contracts.

Reply:
I don't have a very strong opinion though I still lean towards renaming. 
Small note on my reasoning why I still lean on renaming - if we at one point find that a different approach is better (renaming, implementation, etc) I would never keep it "wrong" just because it's the same in another place. I'd better setup a task to fix everywhere 

@dragos-rebegea what do you say?